tests/unit/IonParserBinaryRawTest.js

```diff
@@ -182,6 +182,120 @@ define([
     readSignedInt([0x7F, 0xFF, 0xFF, 0xFF, 0xFF], maxValueForBits(39));
     readSignedInt([0xFF, 0xFF, 0xFF, 0xFF, 0xFF], -1 * maxValueForBits(39));
 
+    /**
+     * VarUInt
+     *
+     * Spec: http://amzn.github.io/ion-docs/docs/binary.html#varuint-and-varint-fields
+     */
+
+    // The base test function called by the more specific flavors below.
+    // Creates a test which will attempt to read the `expected` VarUInt from the provided input bytes, handling
+    // any anticipated exceptions.
+    let readVarUnsignedIntTest = function(testName, bytes, expected, throwsException) {
+      let test = function() {
+        let binarySpan = new ion.BinarySpan(new Uint8Array(bytes));
+        let actual = ion.ParserBinaryRaw.readVarUnsignedIntFrom(binarySpan);
+        assert.equal(actual, expected)
+      }
+      registerTest(testName, test, throwsException);
+    }
+
+    // Should read the `expected` VarUInt value from the provided input bytes.
+    // Will fail if an exception is thrown or if the value that's read from the bytes is not equal to `expected`.
+    let readVarUnsignedInt = function(bytes, expected) {
+      let testName = 'Read var unsigned int ' + expected + ' from bytes: ' + bytes;
+      let testThrows = false;
+      readVarUnsignedIntTest(testName, bytes, expected, testThrows);
+    }
+
+    // Should detect that overflow has occurred while reading and throw an exception.
+    // Will fail if no exception is thrown.
+    let overflowWhileReadingVarUnsignedInt = function(bytes, expected) {
+      let testName = 'Overflow while attempting to read var unsigned int ' + expected + ' from the bytes: ' + bytes;
+      let testThrows = true;
+      readVarUnsignedIntTest(testName, bytes, expected, testThrows);
```

Comment:
Is overflow the only reason that can cause the call to throw?

Reply:
As written, yes.